### PR TITLE
community[minor]: feat: add chat models for zhipuai

### PR DESCRIPTION
**Pull request by lukywong**
Fixes # (issue)

### CODE REVIEW

File: langchain/package.json [hunk 1]
```diff
@@ -1335,6 +1335,7 @@
     "ignore": "^5.2.0",
     "ioredis": "^5.3.2",
     "jsdom": "*",
+    "jsonwebtoken": "^9.0.2",
```
Comment (jacoblee93) [inline langchain/package.json:1338]: I think this is a mistake? Can you remove?

File: libs/langchain-community/src/chat_models/zhipuai.ts [hunk 2]
```diff
@@ -0,0 +1,444 @@
+import { BaseChatModel, type BaseChatModelParams } from "@langchain/core/language_models/chat_models";
+import { AIMessage, type BaseMessage, ChatMessage } from "@langchain/core/messages";
+import { type ChatResult } from "@langchain/core/outputs";
+import { type CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as jwt from 'jsonwebtoken';
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/zhipuai.ts:7]: Can we add this as a peer dependency instead?

https://github.com/langchain-ai/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md#third-party-dependencies

File: libs/langchain-community/src/chat_models/zhipuai.ts [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+  output: {
+    text: string;
+    finish_reason: "stop" | "length" | "null" | null;
+  };
+}
+
+/**
+ * Interface defining the input to the ZhipuAIChatInput class.
+ */
+interface ZhipuAIChatInput {
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/zhipuai.ts:71]: Would be good to export this type - also maybe call `ChatZhipuAIParams`?

File: libs/langchain-community/src/chat_models/zhipuai.ts [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+    fields: Partial<ZhipuAIChatInput> & BaseChatModelParams = {}
+  ) {
+    super(fields);
+
+    this.zhipuAIApiKey = ChatZhipuAI.encodeApiKey(fields?.zhipuAIApiKey ?? getEnvironmentVariable("ZHIPUAI_API_KEY"));
+    if (!this.zhipuAIApiKey) {
+      throw new Error("ZhipuAI API key not found");
+    }
+
+    this.apiUrl = "https://open.bigmodel.cn/api/paas/v4/chat/completions";
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/zhipuai.ts:198]: Can you link me to docs verifying this URL is correct?

File: libs/langchain-community/src/chat_models/zhipuai.ts [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    this.lc_serializable = true;
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/zhipuai.ts:199]: You don't need to set this here

File: libs/langchain-community/src/chat_models/zhipuai.ts (hunk 2 above)
Comment (lukywong) [reply libs/langchain-community/src/chat_models/zhipuai.ts:7]: > Can we add this as an optional peer dependency instead?
> 
> https://github.com/langchain-ai/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md#third-party-dependencies

yes, thanks for checking, please help to check this again

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**